src/ucp/core/ucp_ep.inl

```diff
@@ -219,4 +219,13 @@ static inline ucp_lane_index_t ucp_ep_get_cm_lane(ucp_ep_h ep)
     return ucp_ep_config(ep)->key.cm_lane;
 }
 
+static UCS_F_ALWAYS_INLINE int
+ucp_ep_send_is_inline(ucp_ep_h ep, const ucp_memtype_thresh_t *max_short,
+                      ssize_t length)
+{
+    return (ucs_likely(length <= max_short->memtype_off) ||
+            (length <= max_short->memtype_on &&
+             ucp_memory_type_cache_is_empty(ep->worker->context)));
```

Comment:
i think we said ucp_memory_type_cache_is_empty() is not really a good test because we are adding the UNKNOWN regions.. so better remote it instead of having more logic depend on it
need just one max_short. it will be a real number of no mem_type_mds, and -1 if there is at least 1 mem type md

Reply:
yes
checking `ucs_likely(length <= max_short->memtype_off)` would be enough

Reply:
we don't need to add memtype on/off for rma
keep one max_short as it was before

Reply:
> we don't need to add memtype on/off for rma

okay, I see now
btw, do we need the same for TAG-matching proto? we could remove memtype on/off for it as well and use `max_short` that will be `-1` in case of there is at least one mem_type MD

Reply:
yes, we talked about removing it for tag matching

Reply:
> yes, we talked about removing it for tag matching

let's remove it in this PR? wdyt?

Reply:
better not: let's keep this PR focused on RMA memtype handling, and cleanup in another PR